packages/fxa-payments-server/src/components/AppLayout/index.tsx

```diff
@@ -10,54 +11,75 @@ export type AppLayoutProps = {
 export const AppLayout = ({ children }: AppLayoutProps) => {
   const { config } = useContext(AppContext);
 
-  return <>
-    <div id="stage" data-testid="stage" className="fade-in-forward" style={{ opacity: 1 }}>
-      {children}
-    </div>
-    <footer data-testid="footer">
-      <div id="about-moz-footer" data-testid="about-moz-footer">
-        <a
-          id="about-mozilla"
-          rel="author noopener noreferrer"
-          target="_blank"
-          href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"
-        >
-          &nbsp;
-        </a>
-      </div>
-      <div id="legal-footer" data-testid="legal-footer">
-        <a className="terms"
-          rel="noopener noreferrer"
-          target="_blank"
-          href={config.legalDocLinks.termsOfService}>
-          Terms of Service
-        </a>
-        <a className="privacy"
-          rel="noopener noreferrer"
-          target="_blank"
-          href={config.legalDocLinks.privacyNotice}>
-          Privacy Notice
-        </a>
+  return (
+    <>
+      <div
+        id="stage"
+        data-testid="stage"
+        className="fade-in-forward"
+        style={{ opacity: 1 }}
+      >
+        {children}
       </div>
-    </footer>
-  </>;
+      <footer data-testid="footer">
+        <div id="about-moz-footer" data-testid="about-moz-footer">
+          <a
+            id="about-mozilla"
+            rel="author noopener noreferrer"
+            target="_blank"
+            href="https://www.mozilla.org/about/?utm_source=firefox-accounts&amp;utm_medium=Referral"
+          >
+            &nbsp;
+          </a>
+        </div>
+        <div id="legal-footer" data-testid="legal-footer">
+          <a
+            className="terms"
+            rel="noopener noreferrer"
+            target="_blank"
+            href={config.legalDocLinks.termsOfService}
+          >
+            Terms of Service
+          </a>
+          <a
+            className="privacy"
+            rel="noopener noreferrer"
+            target="_blank"
+            href={config.legalDocLinks.privacyNotice}
+          >
+            Privacy Notice
+          </a>
+        </div>
+      </footer>
+    </>
+  );
 };
 
 export type SignInLayout = {
   children: ReactNode;
 };
 
-export const SignInLayout = ({ children }: SignInLayout) => (
-  <>
+export const SignInLayoutContext = React.createContext({
+  setHideLogo: (hideLogo: boolean) => {},
```

Comment:
Looks like we just have one spot where we want to hide this logo - i.e. when the profile avatar appears. So, I figured that threading through a setState via context was the least-worst way to get a signal from children